### PR TITLE
Add rollback diff to full sync CI

### PR DESCRIPTION
**Pull request by Jouzo**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

- Output all the currently log command to a debug-pre-rollback.log file at START_BLOCK
- Syncs the 50k range
- Rollbacks to START_BLOCK and re-run all the log command and outputs it to debug-post-rollback.log
- Add a CI step where we diff debug-pre-rollback.log against debug-pot-rollback.log

#### Which issue(s) does this PR fixes?:

Would have caught re-org issue fixed in https://github.com/DeFiCh/ain/pull/1641

See run https://github.com/DeFiCh/ain/actions/runs/4058261206/jobs/6985038469#step:10:20 with revert of PR 1641 
and run https://github.com/DeFiCh/ain/actions/runs/4058262361/jobs/6985091099#step:10:20 with PR 1641

### CODE REVIEW

File: .github/workflows/fullsync-tests.yml [hunk 1]
```diff
@@ -129,10 +129,17 @@ jobs:
...
-      run: ./sync_then_diff.sh
+    - name: Sync
+      run: ./sync.sh
+
+    - name: Diff log
+      run: diff debug-${{matrix.blocks.stop}}.log debug-tmp-${{matrix.blocks.stop}}.log
+
+    - name: Diff rollback log
+      run: diff debug-pre-rollback.log debug-post-rollback.log
+      if: ${{ failure() ||  success() }}
```
Comment (prasannavl) [inline .github/workflows/fullsync-tests.yml:142]: Need this?

**Comment by prasannavl** [comment]
This is neat. 

A few questions:

- Could we avoid an `invalidateblock` in the beginning, even if it's the same start block?
- Don't recall the cause for the edge case of block being stuck and reinvalidation required. Wonder if it's related to #1700 - if it is, we can clean up in addition to this.

File: .github/workflows/fullsync-tests.yml (hunk 1 above)
Comment (Jouzo) [reply .github/workflows/fullsync-tests.yml:142]: Not strictly needed but let's it run even if we have a failure on previous diff log.
It let's us have both diff results but will still mark the sync as failed from previous diff failure (whereas `continue-on-error` would mark it as passed).

**Comment by Jouzo** [comment]
I will do a run without the `reconsiderblock` and check if it has been fixed by #1700

**Review by prasannavl — APPROVED** [review]
